--- a/subprojects/singularity-write/src/ui/doc_view.vala
+++ b/subprojects/singularity-write/src/ui/doc_view.vala
@@ -1904,5 +1904,27 @@
             attribute_values = {};
             return false;
         }
+
+        public void get_default_attributes(out string[] attribute_names, out string[] attribute_values) {
+            attribute_names = {};
+            attribute_values = {};
+        }
+
+        public bool get_extents(uint start, uint end, Graphene.Rect extents) {
+            return false;
+        }
+
+        public bool get_offset(Graphene.Point point, out uint offset) {
+            offset = 0;
+            return false;
+        }
+
+        public bool set_caret_position(uint offset) {
+            return false;
+        }
+
+        public bool set_selection(size_t i, Gtk.AccessibleTextRange range) {
+            return false;
+        }
     }
 }
